src/main/java/io/jsonwebtoken/impl/DefaultJwtParser.java

```diff
@@ -309,6 +363,47 @@ public Jwt parse(String jwt) throws ExpiredJwtException, MalformedJwtException,
         }
     }
 
+    private void validateExpectedClaims(Header header, Claims claims) {
+        for (String expectedClaimName : expectedClaims.keySet()) {
+            Object expectedClaimValue;
+            Object actualClaimValue;
+
+            // since issued at is a date, call the specific method
+            // other methods deal with strings and the more
+            // general method can be used
```

Comment:
Right. I get that. I was just trying to avoid having a ton of branching logic. I was thinking of having the registered claim types as an enum instead of constants so we could `switch` on it or using reflection, but that seemed like overkill.